aws/resource_aws_kinesis_analyticsv2_application_test.go

```diff
@@ -0,0 +1,1631 @@
+	return fmt.Sprintf(`
+resource "aws_kinesis_stream" "test" {
+  name        = "testAcc-%s-%d"
+  shard_count = 1
+}
+
+resource "aws_kinesis_analyticsv2_application" "test" {
+  name                   = "testAcc-%d"
+  code                   = "testCode\n"
+  runtime                = "SQL-1_0"
+  code_content_type      = "plain_text"
+  service_execution_role = "${aws_iam_role.test.arn}"
+
+  sql_application_configuration {
+    outputs {
+      name = "test_name2"
+
+      kinesis_stream {
+        resource_arn = "${aws_kinesis_stream.test.arn}"
+      }
+
+      schema {
+        record_format_type = "CSV"
+      }
+    }
+  }
+}
+`, streamName, rInt, rInt)
+}
+
+func testAccKinesisAnalyticsV2Application_referenceDataSource(rInt int) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket" "test" {
+  bucket = "testacc-%d"
+}
+
+resource "aws_kinesis_analyticsv2_application" "test" {
+  name                   = "testAcc-%d"
+  runtime                = "SQL-1_0"
+  code                   = "testCode\n"
+  code_content_type      = "plain_text"
+  service_execution_role = "${aws_iam_role.test.arn}"
+
+  sql_application_configuration {
+    reference_data_sources {
+      table_name = "test_table"
+
+      s3 {
+        bucket_arn = "${aws_s3_bucket.test.arn}"
+        file_key   = "test_file_key"
+      }
+
+      schema {
+        record_columns {
+          mapping  = "$.test"
+          name     = "test"
+          sql_type = "VARCHAR(8)"
+        }
+
+        record_encoding = "UTF-8"
+
+        record_format {
+          mapping_parameters {
+            json {
+              record_row_path = "$"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+`, rInt, rInt)
+}
+
+func testAccKinesisAnalyticsV2Application_referenceDataSourceUpdate(rInt int) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket" "test" {
+  bucket = "testacc2-%d"
+}
+
+resource "aws_kinesis_analyticsv2_application" "test" {
+  name                   = "testAcc-%d"
+  runtime                = "SQL-1_0"
+  code                   = "testCode\n"
+  code_content_type      = "plain_text"
+  service_execution_role = "${aws_iam_role.test.arn}"
+
+  sql_application_configuration {
+    reference_data_sources {
+      table_name = "test_table2"
+
+      s3 {
+        bucket_arn = "${aws_s3_bucket.test.arn}"
+        file_key   = "test_file_key"
+      }
+
+      schema {
+        record_columns {
+          mapping  = "$.test2"
+          name     = "test2"
+          sql_type = "VARCHAR(8)"
+        }
+
+        record_encoding = "UTF-8"
+
+        record_format {
+          mapping_parameters {
+            csv {
+              record_column_delimiter = ","
+              record_row_delimiter    = "\n"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+`, rInt, rInt)
+}
+
+// this is used to set up the IAM role
+func testAccKinesisAnalyticsV2Application_prereq(rInt int) string {
```

Comment:
@blake-wilson We've recently updated [guidance](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#base-test-configurations) on acceptance tests and prefer naming a function such as this `testAccKinesisAnalyticsV2ApplicationConfigBase()` and using that, in conjunction with `composeConfig` in each config generator rather than explicitly using it in each test setup.